src/index.ts

```diff
@@ -12,6 +12,36 @@ export * from "./exporters";
 export * from "./processors";
 export * from "./api";
 
+/**
+ * KnownString enforces that:
+ *
+ * - T must be a string
+ * - string must NOT be a T
+ *
+ * This effectively requires T to NOT be an arbitrary string - it must be
+ * a string value known ahead of time.
+ */
+type KnownString<T extends string> = string extends T
+  ? "INPUT TYPE ERROR: string type is too broad, should be a known value"
+  : T;
+
+/**
+ * KnownKeys enforces that:
+ *
+ * - T must be an object with string keys (KeyT) and number values
+ * - an object with arbitary strings as keys must NOT be a T
+ *
+ * This effectively requires KeyT to NOT be an arbitrary string - keys must be
+ * known ahead of time.
+ */
+type KnownKeys<KeyT extends string, T extends { [key in KeyT]: number }> = {
```

Comment:
Minor note: in TS, we usually do not suffix types with `T`. A regular `Key extends string` will do that.